ports/libsimpleservo/src/api.rs

```diff
@@ -0,0 +1,418 @@
+    pub static SERVO: RefCell<Option<ServoGlue>> = RefCell::new(None);
+}
+
+/// The EventLoopWaker::wake function will be called from any thread.
+/// It will be called to notify embedder that some events are available,
+/// and that perform_updates need to be called
+pub use servo::embedder_traits::EventLoopWaker;
+
+/// Delegate resource file reading to the embedder.
+pub trait ReadFileTrait {
+    fn readfile(&self, file: &str) -> Vec<u8>;
+}
+
+/// Callbacks. Implemented by embedder. Called by Servo.
+pub trait HostTrait {
+    /// Will be called from the thread used for the init call.
+    /// Will be called when the GL buffer has been updated.
+    fn flush(&self);
+    /// Page starts loading.
+    /// "Reload button" should be disabled.
+    /// "Stop button" should be enabled.
+    /// Throbber starts spinning.
+    fn on_load_started(&self);
+    /// Page has loaded.
+    /// "Reload button" should be enabled.
+    /// "Stop button" should be disabled.
+    /// Throbber stops spinning.
+    fn on_load_ended(&self);
+    /// Page title has changed.
+    fn on_title_changed(&self, title: String);
+    /// Page URL has changed.
+    fn on_url_changed(&self, url: String);
+    /// Back/forward state has changed.
+    /// Back/forward buttons need to be disabled/enabled.
+    fn on_history_changed(&self, can_go_back: bool, can_go_forward: bool);
+    /// Page animation state has changed. If animating, it's recommended
+    /// that the embedder doesn't wait for the wake function to be called
+    /// to call perform_updates. Usually, it means doing:
+    /// while true { servo.perform_updates() }. This will end up calling flush
+    /// which will call swap_buffer which will be blocking long enough to limit
+    /// drawing at 60 FPS.
+    /// If not animating, call perform_updates only when needed (when the embedder
+    /// has events for Servo, or Servo has woken up the embedder event loop via
+    /// EventLoopWaker).
+    fn on_animating_changed(&self, animating: bool);
+}
+
+pub struct ServoGlue {
+    servo: Servo<ServoCallbacks>,
+    batch_mode: bool,
+    callbacks: Rc<ServoCallbacks>,
+    browser_id: BrowserId,
+    events: Vec<WindowEvent>,
+    current_url: Option<ServoUrl>,
+}
+
+pub fn servo_version() -> String {
+    servo::config::servo_version()
+}
+
+/// Initialize Servo. At that point, we need a valid GL context.
+/// In the future, this will be done in multiple steps.
+pub fn init(
+    gl: Rc<gl::Gl>,
+    argsline: String,
+    embedder_url: Option<String>,
+    waker: Box<EventLoopWaker>,
+    readfile: Box<ReadFileTrait + Send + Sync>,
+    callbacks: Box<HostTrait>,
+    width: u32,
+    height: u32,
+) -> Result<(), &'static str> {
+    resources::set(Box::new(ResourceReader(readfile)));
+
+    // opts::from_cmdline_args expects the first argument to be the binary name.
+    let argsline = format!("servo {}", argsline);
+    let args: Vec<String> = argsline.split_whitespace().map(|s| s.to_owned()).collect();
```

Comment:
`split_whitespace` accepts a number of characters, including some non-ASCII, as separators. This should at least be `.split(' ')`, but see other comment.